server_src/runners/history.js

```diff
@@ -327,32 +617,42 @@ class HistoryCollection {
     }
   }
 
-  /**
-   * Grabs the timelapse by downloading it from OctoPrint's API
-   * @param fileName
-   * @param url
-   * @param id
-   * @param printer
-   * @param serverSettings
-   * @returns {Promise<string>}
-   */
   static async grabTimeLapse(fileName, url, id, printer, serverSettings) {
+    const download = async (url, path, callback) => {
+      const res = await fetch(url, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+          "X-Api-Key": printer.apikey,
+        },
+      });
+      const fileStream = fs.createWriteStream(path);
+      await new Promise((resolve, reject) => {
+        res.body.pipe(fileStream);
+        res.body.on("error", reject);
+        fileStream.on("finish", callback);
+      });
+    };
 
     const path = `./images/historyCollection/timelapses/${id}-${fileName}`;
-    if (!fs.existsSync("./images/historyCollection")) {
-      fs.mkdirSync("./images/historyCollection");
+    if (!fs.existsSync(`./images/historyCollection`)) {
+      fs.mkdirSync(`./images/historyCollection`);
     }
-    if (!fs.existsSync("./images/historyCollection/timelapses")) {
-      fs.mkdirSync("./images/historyCollection/timelapses");
+    if (!fs.existsSync(`./images/historyCollection/timelapses`)) {
+      fs.mkdirSync(`./images/historyCollection/timelapses`);
     }
 
-    await downloadFromOctoPrint(url, path, () => {
+    download(url, path, () => {
```

Comment:
Ok the merge has failed. This is not ok as it is reversing my and your code haha. Ill stop the review here

Reply:
That's because I hadn't actually gotten to the download function yet, loads to nuke still. I'll be updating it to your download function for images and timelapse